django/utils/numberformat.py

```diff
@@ -1,7 +1,7 @@
+import builtins
 from decimal import Decimal
 
 from django.conf import settings
-from django.utils.safestring import mark_safe
```

Comment:
Hi @tim-mccurrach I'm inclined to agree, doesn't look like a major speed bump to revert this, since both `mark_safe` and `_format_int` have to evaluate a couple statements before casting to str. Your view on this?

Reply:
Ah, sorry, I see that your reply is found in the comment below this review. I see your view, but it seems to me this is a change in the underlying functionality, not just optimizing for speed. So it could slow down reviews/require a release note. Isn't it possible someone has code taking the result of numberformat.format and checking whether or not it's a SafeString before modifying it?

Reply:
I've shaved a little bit off by returning a `SafeString` directly instead of calling `mark_safe`. It still is quite a bit slower for integers:
```
old_format:                       Mean +- std dev: 4.71 us +- 0.18 us
new_format(with safe strings):    Mean +- std dev: 4.30 us +- 0.22 us
new_format(without safe strings): Mean +- std dev: 3.11 us +- 0.12 us
``` 
Whilst it would be a breaking change, the current behaviour definitely seems like the 'wrong' behaviour. Nonetheless, in the interest of getting this ticket moving, I've made the change. It can always be addressed in a follow up ticket.